### PR TITLE
initial version of LSTM character LM

### PR DESCRIPTION
**Pull request by pbrakel**
I'm still working on notebooks and might make some slides but I figured you might want to test this experiment in quicklab to see if it runs because it will be one that would benefit from a GPU and I don't expect I'll introduce additional dependencies. I also haven't verified the correctness of the mask usage yet but the network seems to generate somewhat decent looking text after a while (depending on the settings).

### CODE REVIEW

**Comment by carriepl** [comment]
Is there any way to get a more objective performance measure (like an error value on some validation set or anything). This would make it more interesting to people who want to tinker with the model see the effect of new stuff (adadelta, rmsprop, more layers, different parameters, ...). Just looking at some generated text will make it hard for people to judge the effect of their modifications (or even just do early stopping, really).

**Comment by pbrakel** [comment]
Thanks for checking the code! This should be very easy to add by adding a separate file with evaluation text to compute the total cross-entropy/perplexity/bits-per-character for. I didn't do it yet because it will probably be quite slow but I could keep the validation set very small. I don't think people will be able to run it for long enough to make early stopping very interesting though.

**Comment by pbrakel** [comment]
I added a notebook that should download and unzip the data. The code now also reports validation error scores.

**Comment by pbrakel** [comment]
I fixed the BFGS issue.

**Comment by lamblin** [comment]
Thanks, it seems to work!
